evap/staff/tests/test_views.py

```diff
@@ -185,64 +187,137 @@ def setUpTestData(cls):
         semester = baker.make(Semester, participations_are_archived=True)
-        evaluation = baker.make(Evaluation, course=baker.make(Course, semester=semester), _participant_count=0, _voter_count=0)
+        evaluation = baker.make(Evaluation,
+            course=baker.make(Course, semester=semester), _participant_count=0, _voter_count=0
+        )
         contribution2 = baker.make(Contribution, evaluation=evaluation)
-        baker.make(UserProfile, username='contributor1', contributions=[contribution1])
-        baker.make(UserProfile, username='contributor2', contributions=[contribution2])
+        baker.make(UserProfile, email='contributor1@institution.example.com', contributions=[contribution1])
+        baker.make(UserProfile, email='contributor2@institution.example.com', contributions=[contribution2])
 
-        page = self.app.get(self.url, user='manager')
-        form = page.forms["user-bulk-delete-form"]
+        expected_users = set(UserProfile.objects.all())
+        expected_users.discard(UserProfile.objects.get(email='testuser2@institution.example.com'))
 
+        page = self.app.get(self.url, user='manager')
+        form = page.forms["user-bulk-update-form"]
         form["username_file"] = (self.filename,)
+        response = form.submit(name="operation", value="test")
 
-        user_count_before = UserProfile.objects.count()
-
-        reply = form.submit(name="operation", value="bulk_delete")
+        self.assertIn(
+            '1 will be updated, 1 will be deleted and 1 will be marked inactive. 1 new users will be created.',
+            response
+        )
+        self.assertIn('testupdate@institution.example.com > testupdate@internal.example.com', response)
 
-        # Getting redirected after.
-        self.assertEqual(reply.status_code, 302)
+        form = response.forms["user-bulk-update-form"]
+        response = form.submit(name="operation", value="bulk_update")
 
-        # Assert only one user got deleted and one was marked inactive
-        self.assertTrue(UserProfile.objects.filter(username='testuser1').exists())
-        self.assertFalse(UserProfile.objects.filter(username='testuser2').exists())
+        # testuser1 is in the file and must not be deleted
+        self.assertTrue(UserProfile.objects.filter(email='testuser1@institution.example.com').exists())
+        # testuser2 is not in the file and must be deleted
+        self.assertFalse(UserProfile.objects.filter(email='testuser2@institution.example.com').exists())
+        # manager is not in the file but still must not be deleted
         self.assertTrue(UserProfile.objects.filter(username='manager').exists())
+        # testusernewinternal is a new internal user and should be created
+        expected_users.add(UserProfile.objects.get(email='testusernewinternal@institution.example.com'))
+        self.assertTrue(UserProfile.objects.filter(email='testusernewinternal@institution.example.com').exists())
```

Comment:
This assert should come before the `get` one line above, otherwise the test will error via the get instead of fail via the assert